frontend/src/entry/model/releases/AssociatedReleasesDialog.tsx

```diff
@@ -0,0 +1,96 @@
+import { Dialog, DialogContent, DialogTitle, List, ListItem, ListItemText, Stack, Typography } from '@mui/material'
+import { useGetModel } from 'actions/model'
+import { useGetReleasesForModelId } from 'actions/release'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import EmptyBlob from 'src/common/EmptyBlob'
+import Loading from 'src/common/Loading'
+import Link from 'src/Link'
+import MessageAlert from 'src/MessageAlert'
+import { FileInterface, isFileInterface, ReleaseInterface } from 'types/types'
+import { formatDateString } from 'utils/dateUtils'
+
+type AssociatedReleasesDialogProps = {
+  modelId: string
+  file: FileInterface | File
+  open: boolean
+  onClose: () => void
+}
+
+export default function AssociatedReleasesDialog({ modelId, file, open, onClose }: AssociatedReleasesDialogProps) {
+  const { releases, isReleasesLoading, isReleasesError } = useGetReleasesForModelId(modelId)
+  const { model, isModelLoading, isModelError } = useGetModel(modelId, 'model')
+  const [latestRelease, setLatestRelease] = useState('')
+
+  const associatedReleases: Array<ReleaseInterface> = useMemo(
+    () => releases.filter((release) => isFileInterface(file) && release.fileIds.includes(file._id)),
+    [file, releases],
+  )
+
+  const sortAssociatedReleases: Array<ReleaseInterface> = associatedReleases.sort((a, b) => {
```

Comment:
This could use the utility function I've suggested you create in `frontend/utils/arrayUtils.ts`.

We can also combine it with the above `associatedReleases` logic, so we don't have multiple arrays to deal with.
```
  const associatedReleases: Array<ReleaseInterface> = useMemo(
    () => releases.filter((release) => isFileInterface(file) && release.fileIds.includes(file._id)).sort(sortByCreatedAtDescending),
    [file, releases],
  )
```